src/main/scala/org/constellation/primitives/TransactionGenerator.scala

```diff
@@ -128,21 +128,32 @@ class TransactionGenerator[F[_]: Concurrent](
     addresses: Seq[(String, Option[AddressCacheData])],
     peers: Seq[(Id, PeerData)]
   ): F[Transaction] =
-    transactionService
-      .createTransaction(dao.selfAddressStr, randomAddressFrom(addresses), randomAmount(rangeAmount), dao.keyPair)
+    transactionService.transactionChainService.createAndSetLastTransaction(
+    dao.selfAddressStr,
+    randomAddressFromPeers(peers),
+    randomAmount(rangeAmount),
+    dao.keyPair,
+      false
+  )
 
   private def generateMultipleAddressTransactionWithoutSufficent(
     addresses: Seq[(String, Option[AddressCacheData])],
     peers: Seq[(Id, PeerData)]
   ): F[Transaction] = {
-    val source = getSourceAddressForTxWithoutSufficient(addresses)
-    transactionService.createTransaction(
-      source,
-      randomAddressFromPeers(peers),
-      randomAmount(rangeAmount),
-      keyPairForSource(source),
-      normalized = false
-    )
+    val source: String = getSourceAddressForTxWithoutSufficient(addresses)
+    transactionService.transactionChainService.getLastTransactionRef(dao.selfAddressStr)
+      .flatMap { ref: LastTransactionRef =>
```

Comment:
concurrency issue

Reply:
Missed this when refactoring, should be fixed above and elsewhere